src/components/HourlyCityWeather/HourlyCityWeather.js

```diff
@@ -0,0 +1,168 @@
+import { useEffect, useState } from "react";
+import { Col, Row } from "react-bootstrap";
+import Carousel from 'react-bootstrap/Carousel';
+import './HourlyCityWeather.css';
+
+const HourlyCityWeather = ({city}) => {
+    const [error, setError] = useState(null);
+    const [isLoaded, setIsLoaded] = useState(false);
+    const [timeStamps, setTimeStamps] = useState(0);
+
+
+    useEffect(() => {
+        fetch(
+            "https://api.openweathermap.org/data/2.5/forecast?q=" +
+            city +
+            "&units=metric" +
+            "&appid=" +
+            process.env.REACT_APP_APIKEY
+        )
+            .then((res) => res.json())
+            .then(
+                (result) => {
+                    if (result["cod"] !== "200") {
+                        setIsLoaded(true);
+                        setError(result);
+                    } else {
+                        setIsLoaded(true);
+                        setError();
+                        organizeForecast(result);
+                    }
+                },
+                (error) => {
+                    setError(error);
+                }
+            )
+            .catch((err) => {
+                setError(err);
+            });
+
+    }, [city], []);
+
+    function organizeForecast(result) {
+        const hourlyForecasts = [];
+        var dailyData = [];
+        for (let i = 0; i < result.list.length; i++) {
+            var dateAndTime = result.list[i].dt_txt.split(" ");
+            var time = dateAndTime[1].split(":");
+            time = time[0] + ":" + time[1];
+            dailyData = {
+                "day": getWeekday(result.list[i].dt_txt),
+                "time": time,
+                "temperature": parseInt(result.list[i].main.temp, 10),
+                "icon": result.list[i].weather[0].main
+            };
+            hourlyForecasts.push(dailyData);
+        }
```

Comment:
Rather than using a for-loop to populate a list, you can use the `.map` method.

```suggestion
        const hourlyForecasts = result.list.map((item) => ({
          day: getWeekday(item.dt_txt),
          time: item.dt_txt.slice(-8, -3), 
          temperature: parseInt(item.main.temp, 10),
          icon: item.weather[0].main
        }))
```

Since we know the time is 0 padded it's quicker just to slice the string with `.slice(-8, -3)`. Or you could move that to a helper function like `getWeekday`.

Reply:
It looks way better this way.